charts/generic-prometheus-alerts/templates/aws-elasticache-alerts.yaml

```diff
@@ -22,15 +22,15 @@ spec:
   groups:
     - name: {{ $targetApplication }}-elasticache
       rules:
-        {{- range $clusterId := $.Values.elastiCacheAlertsClusterIds }}
+        {{- range $clusterId, $displayName := $.Values.elastiCacheAlertsClusterIds }}
         - alert: elasticache-enginecpu-utilisation
           annotations:
             message: elasticache cluster {{ $clusterId }} in {{ $targetNamespace }} - EngineCPU utilisation at {{`{{`}} printf "%.2f" $value {{`}}`}} which is over threshold of {{ $elastiCacheAlertsEngineCPUThreshold }} for last {{ $elastiCacheAlertsEngineCPUThresholdMinutes }} mins. This alert configured by app {{ $targetNamespace }}/{{ $targetApplication }}.
             runbook_url: {{ $.Values.runbookUrl }}elasticache-enginecpu-utilisation
             dashboard_url: {{ $.Values.grafanaUrl }}/d/nK7rpiQZk/aws-elasticache-redis?orgId=1&var-cacheclusterId={{ $clusterId }}&from=now-6h&to=now
             summary: elasticache cluster EngineCPU usage too high
           expr: |-
-            (sum by (cache_cluster_id) (aws_elasticache_engine_cpuutilization_average{cache_cluster_id='{{ $clusterId }}'} offset {{ $elastiCacheAlertsEngineCPUThresholdMinutes }}m) > {{ $elastiCacheAlertsEngineCPUThreshold }})
+            aws_elasticache_engine_cpuutilization_average{cache_cluster_id='{{ $clusterId }}'} offset {{ $elastiCacheAlertsEngineCPUThresholdMinutes }}m > {{ $elastiCacheAlertsEngineCPUThreshold }}
```

Comment:
I didn't understand why we were doing this sum by... Prometheus spits out two results for the query, one with and one without a `cache_node_id="0001"`.  So the sum by was artificially doubling the values in the check!

Reply:
suspect this was meant to take into account the entire cluster, if for instance there are more than one node.